proc/combiner.go

```diff
@@ -0,0 +1,125 @@
+package proc
```

Comment:
I put this in proc for now, but in a follow-on PR I think we could replace zbuf.Combiner with it. This could also be used by sort.